format/mbr/mbr.go

```diff
@@ -0,0 +1,66 @@
+package mbr
+
+import (
+	"fmt"
+
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/format/registry"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+func init() {
+	registry.MustRegister(decode.Format{
+		Name:        format.MBR,
+		Description: "Master Boot Record",
+		DecodeFn:    mbrDecode,
+	})
+}
+
+func decodePartitionTableEntry(d *decode.D) {
+	d.FieldU8("boot_indicator", scalar.UToDescription{
+		0x80: "active",
+		0x00: "inactive",
+	})
+	d.FieldStrScalarFn("starting_chs_vals", decodeCHSBytes)
+	d.FieldU8("partition_type", partitionTypes)
+	d.FieldStrScalarFn("ending_chs_vals", decodeCHSBytes)
+	d.FieldStrScalarFn("starting_sector", decodeCHSBytes)
+	d.U8() // extra byte
+	d.FieldScalarU32("partition_size")
+}
+
+// Because this is a fixed-sized table, I am opting to use a
+// FieldStruct instead of a FieldArray
+func decodePartitionTable(d *decode.D) {
+	d.FieldStruct("entry_1", decodePartitionTableEntry)
+	d.FieldStruct("entry_2", decodePartitionTableEntry)
+	d.FieldStruct("entry_3", decodePartitionTableEntry)
+	d.FieldStruct("entry_4", decodePartitionTableEntry)
+}
+
+// Source: https://thestarman.pcministry.com/asm/mbr/PartTables.htm#Decoding
+func decodeCHSBytes(d *decode.D) scalar.S {
+	head, _ := d.Bits(8)
+	sectorHighBits, err := d.Bits(2)
+	if err != nil {
+		d.IOPanic(err, "chs")
+	}
+	sector, _ := d.Bits(6)
+	cylinderLowerBits, err := d.Bits(8)
+	if err != nil {
+		d.IOPanic(err, "chs")
+	}
+	cylinder := (sectorHighBits << 2) | cylinderLowerBits
+	return scalar.S{Actual: fmt.Sprintf("CHS(%x, %x, %x)", cylinder, head, sector)}
+}
```

Comment:
Could it make sense to read CHS as a struct with cylinder, head , sector fields?

Reply:
Yes, and that suggestion is in line with [this general tip](https://github.com/wader/fq/blob/master/doc/dev.md#some-general-tips:~:text=Try%20to%20not%20decode%20too%20much%20as%20one%20value.%20A%20length%20encoded%20int%20could%20be%20two%20fields%2C%20but%20maybe%20a%20length%20prefixed%20string%20should%20be%20one.%20Flags%20can%20be%20struct%20with%20bit%2Dfields.)

Reply:
Ah yes, forgot i wrote that :) maybe should add something more about that it also makes it easier to write queries if things are split up into more individual fields